### ISSUE TITLE
[BUG] generic cannot be used on left side of < and > operators 

### ISSUE THREAD

**Issue by Fusioon** [beefytech/Beef#1143]
Code to reproduce:
`````` 
class Foo<T>
{
    public static T value;
}

class Program
{
    static void Main()
    {
        let foo = Foo <int>.value;
        bool works1 = foo > Foo<int>.value;
        bool works2 = foo < Foo<int>.value;
        bool doestnWork1 = Foo<int>.value < foo;
        bool doestnWork2 = Foo<int>.value > foo;
    }
}
````````
Tested with: ab93df4

**Comment by bfiete** [comment]
One-character typo fix... 90f34b6bee177bc4d1b8a59aca631e045e9e4ade
Thanks, C++, for not even warning on comparing two completely different enum types.

**Comment by bfiete** [comment]
Fixed at e508991dce894ea2a57a2c498c7e00d2ca835227
